libs/ml/include/ml/core/node.hpp

```diff
@@ -149,6 +250,7 @@ class Node
   TensorType        cached_output_;
   CachedOutputState cached_output_status_;
   OpType            operation_type_;
+  ShapeType         default_output_shape_{};
```

Comment:
since you've added a private member variable to Node you will need to update Node serialisation as well (+ add tests to check that this serialisation->deserialisation correctly reproduces this member variable)

Reply:
Got it, ok. I would not do this in this particular PR because it is not going to be merged as is anyway, but yes in the product version I'll add serialisation for new fields in Node and Ops